ui/common/src/device.ts

```diff
@@ -54,14 +54,25 @@ export const isIOS = (constraint?: { below?: number; atLeast?: number }) => {
 
 export const isChrome = (): boolean => /Chrome\//.test(navigator.userAgent);
 
+export const isFirefox = (): boolean => /Firefox/.test(navigator.userAgent);
+
+export const getFirefoxMajorVersion = (): number => {
+  if (!isFirefox()) {
+    return 0;
+  }
+  const match = /Firefox\/(\d*)/.exec(navigator.userAgent);
+  if (!match || match.length < 2) return 0;
+  return parseInt(match[1]);
+};
+
 export const isIOSChrome = (): boolean => /CriOS/.test(navigator.userAgent);
 
 export const isTouchDevice = () => !hasMouse();
 
 export const isIPad = (): boolean =>
   navigator?.maxTouchPoints > 2 && /iPad|Macintosh/.test(navigator.userAgent);
 
-export type Feature = 'wasm' | 'sharedMem' | 'simd';
+export type Feature = 'wasm' | 'sharedMem' | 'simd' | 'dynamicImport';
```

Comment:
Let's call it `webWorkerDynamicImport` to be clear and not confuse with regular dynamic imports. If that's too long for your liking, an abbreviaition such as 'wwdi' is fine too. but in that case, comment the long form at the definition point in device.ts.